mini.go

```diff
@@ -137,19 +143,32 @@ func GET(writer http.ResponseWriter, reader *http.Request) {
 //the function which handle the post method
 //post the json data from broswer to the server and sql databases
 func Poster(w http.ResponseWriter, r *http.Request) {
+	//time when do request
+	RequestStart := time.Now()
 
 	//get the raw bytes of input data
 	bytes, errs := ioutil.ReadAll(r.Body)
 	if errs != nil {
-		fmt.Println(errs)
+		errString := fmt.Sprintf("buffer overflow %s", errs)
+		response(w, errString, "", http.StatusBadRequest)
 		return
 	}
 
-	//store the raw bytes to a temporary struct
+	//store the raw bytes to a temporary struct and log the Json invalid format
 	var point Click
 	errs = json.Unmarshal(bytes, &point)
 	if errs != nil {
-		fmt.Println(errs)
+		errString := fmt.Sprintf("invalid Json format: %s", errs)
+		response(w, errString, "", http.StatusBadRequest)
```

Comment:
same thing here, I don't see anywhere `response()` is defined--- am I missing something?